drivers/SmartThings/zwave-switch/src/test/test_zooz_zen_30_dimmer_relay.lua

```diff
@@ -0,0 +1,723 @@
+        SwitchBinary:Set({
+          target_value = SwitchBinary.value.OFF_DISABLE,
+          duration = 0
+        },
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {0}
+          })
+      )
+    )
+
+    test.wait_for_events()
+    test.mock_time.advance_time(1)
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Get({},
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {0}
+          })
+      )
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "Switch capability on commands should evoke the correct Z-Wave SETs and GETs dst_channel 1",
+  function()
+    test.timer.__create_and_queue_test_time_advance_timer(1, "oneshot")
+    test.socket.capability:__queue_receive({
+      mock_zooz_zen_30_dimmer_relay.id,
+      { capability = "switch", component = "switch1", command = "on", args = {} }
+    })
+
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Set({
+          target_value = SwitchBinary.value.ON_ENABLE,
+          duration = 0
+        },
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {1}
+          })
+      )
+    )
+
+    test.wait_for_events()
+    test.mock_time.advance_time(1)
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Get({},
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {1}
+          })
+      )
+    )
+  end
+)
+
+test.register_coroutine_test(
+  "Switch capability off commands should evoke the correct Z-Wave SETs and GETs dst_channel 1",
+  function()
+    test.timer.__create_and_queue_test_time_advance_timer(1, "oneshot")
+    test.socket.capability:__queue_receive({
+      mock_zooz_zen_30_dimmer_relay.id,
+      { capability = "switch", component = "switch1", command = "off", args = {} }
+    })
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Set({
+          target_value = SwitchBinary.value.OFF_DISABLE
+        },
+        {
+          encap = zw.ENCAP.AUTO,
+          src_channel = 0,
+          dst_channels = {1}
+        }
+        )
+      )
+    )
+    test.wait_for_events()
+    test.mock_time.advance_time(1)
+    test.socket.zwave:__expect_send(
+      zw_test_utils.zwave_test_build_send_command(
+        mock_zooz_zen_30_dimmer_relay,
+        SwitchBinary:Get({},
+          {
+            encap = zw.ENCAP.AUTO,
+            src_channel = 0,
+            dst_channels = {1}
+          })
+      )
+    )
+  end
+)
+
+test.register_message_test(
```

Comment:
@NikitaKuchta Could You also add a test for switch level capability ? (resulting in sending SwitchMultilevel:Set command class)

Reply:
> 

done